core/io.go

```diff
@@ -61,6 +61,12 @@ func (rp *RESPParser) DecodeOne() (interface{}, error) {
 			}
 			return nil, err
 		}
+
+		// Handle the case where no data is read but no error is returned
+		if n == 0 {
+			// This can happen if the connection is closed on the client side but not properly detected
+			return nil, errors.New("ERR possible client-side connection closure")
```

Comment:
IMO, it may be good to return a standard error for example net.ErrClosed and have the capability to have different handling for different error scenarios in the calling function. wdyt?

Reply:
Let's take this up as a part of #225